osquery/tables/system/darwin/crash_logs.cpp

```diff
@@ -0,0 +1,154 @@
+
+/// Locationg of the system application crash logs in OS X
+const std::string kDiagnosticReportsPath = "/Library/Logs/DiagnosticReports";
+/// Map of the values we currently parse out of the log file
+const std::map<std::string, std::string> kCrashDumpKeys = {
+    {"Process", "pid"},
+    {"Path", "path"},
+    {"Log Location", "crash_path"},
+    {"Identifier", "identifier"},
+    {"Version", "version"},
+    {"Parent Process", "parent"},
+    {"Responsible", "responsible"},
+    {"User ID", "user_id"},
+    {"Date/Time", "date_time"},
+    {"Crashed Thread", "crashed_thread"},
+    {"Exception Type", "exception_type"},
+    {"Exception Codes", "exception_codes"},
+    {"Exception Note", "exception_notes"},
+    {"rax", "rax"},
+    {"rbx", "rbx"},
+    {"rcx", "rcx"},
+    {"rdx", "rdx"},
+    {"rdi", "rdi"},
+    {"rsi", "rsi"},
+    {"rbp", "rbp"},
+    {"rsp", "rsp"},
+};
+
+void readCrashDump(const std::string& app_log, Row& r) {
+  r["crash_path"] = app_log;
+  std::string content;
+
+  if (!readFile(app_log, content).ok()) {
+    return;
+  }
+
+  // Variables for capturing the stack trace
+  boost::format crashed_thread_format("Thread %1% Crashed");
+  auto crashed_thread_seen = false;
+
+  auto lines = split(content, "\n");
+  for (auto it = lines.begin(); it != lines.end(); it++) {
+    auto line = *it;
+    // Tokenize first by colons
+    auto toks = split(line, ":");
+
+    if (toks.size() == 0 || kCrashDumpKeys.count(toks[0]) == 0) {
+      continue;
+    }
+
+    // Grab the most recent stack trace line of the crashed thread.
+    if (crashed_thread_seen && toks[0] == crashed_thread_format.str()) {
+      r["stack_trace"] = *(++it);
+      crashed_thread_seen = false;
+    }
+
+    // Process and grab all register values
+    if (toks[0] == "rax" || toks[0] == "rdi") {
+      toks = split(line, ": ");
+      for (size_t i = 0; i < toks.size(); i += 2) {
+        if ((i + 1) < toks.size()) // Ensure that we don't overflow
+          r[kCrashDumpKeys.at(toks[i])] = toks[i + 1];
+      }
+    } else if (toks[0] == "Date/Time") {
+      // Reconstruct split date/time
+      r[kCrashDumpKeys.at(toks[0])] = toks[1] + ":" + toks[2] + ":" + toks[3];
+    } else if (toks[0] == "Crashed Thread") {
+      // If the token is the Crashed thread, update the format string so
+      // we can later grab all of the stack trace
+      r[kCrashDumpKeys.at(toks[0])] = split(toks[1], " ")[0];
+      crashed_thread_format % r[kCrashDumpKeys.at(toks[0])].c_str();
+      crashed_thread_seen = true;
+    } else if (toks[0] == "Process" || toks[0] == "Parent Process") {
+      // Use a regex to extract out the PID value
+      const boost::regex e{"\\[\\d+\\]"};
+      boost::smatch results;
+      if (boost::regex_search(line, results, e)) {
+        auto pid_str = std::string(results[0].first, results[0].second);
+        auto pid = std::stoi(pid_str.substr(1, pid_str.size() - 1));
+        r[kCrashDumpKeys.at(toks[0])] = INTEGER(pid);
```

Comment:
The `INTEGER` macro performs a lexical cast from an `int`-like-type to a `std::string`, if you already have the content in string form as: `substr(pid_str.size() - 1)`, just use that.